tests/testthat/test-get_delay_estimate.R

```diff
@@ -1,124 +1,161 @@
+  {
+    # Add NA in the upper part where it shouldn't be
+    triangle_with_na <- reporting_triangle
+    triangle_with_na[1, 2] <- NA
+
+    expect_error(get_delay_estimate(triangle_with_na))
+  }
+)
+
+test_that("get_delay_estimate errors if not passed a matrix", {
+  # Create a vector instead of a matrix
+  triangle_single_day <- reporting_triangle[1, ]
   expect_error(get_delay_estimate(triangle_single_day))
 })
 
-test_that("get_delay_estimate handles partially complete reporting triangles", {
-  # Test 8: Test that a partial triangle works correctly
-  partial_triangle <- matrix(
-    c(
-      80, 50, 25, 10,
-      100, 50, 30, 20,
-      90, 45, 25, 20,
-      80, 40, 15, NA,
-      70, 30, NA, NA
-    ),
-    nrow = 5,
-    byrow = TRUE
-  )
-  delay_pmf <- get_delay_estimate(
-    reporting_triangle = partial_triangle,
-    max_delay = 3,
-    n = 4
-  )
+test_that("get_delay_estimate calculates correct PMF with complete triangle", {
+  # Create a triangle with known delay PMF
+  complete_pmf <- c(0.45, 0.25, 0.2, 0.1)
+
+  # Generate counts for each reference date
+  complete_counts <- c(80, 100, 90, 80, 70)
+
+  # Create a complete triangle based on the known delay PMF
+  full_triangle <- lapply(complete_counts, function(x) round(x * complete_pmf))
+  full_triangle <- do.call(rbind, full_triangle)
 
-  # Test 9: Test that you get the correct delay with a complete triangle
-  complete_triangle <- matrix(
-    c(
-      80, 50, 25, 10,
-      100, 50, 30, 20,
-      90, 45, 25, 20,
-      80, 40, 15, 5,
-      70, 30, 10, 10
-    ),
-    nrow = 5,
-    byrow = TRUE
-  )
   delay_pmf <- get_delay_estimate(
-    reporting_triangle = complete_triangle,
+    reporting_triangle = full_triangle,
     max_delay = 3,
     n = 5
   )
-  pmf <- colSums(complete_triangle) / sum(complete_triangle)
-  expect_identical(delay_pmf, pmf)
-
-  # Test 10: Test that you get a warning with zeros in the bottom right
-  zero_triangle <- matrix(
-    c(
-      80, 50, 25, 10,
-      100, 50, 30, 20,
-      90, 45, 25, 0,
-      80, 40, 0, 0,
-      70, 0, 0, 0
-    ),
-    nrow = 5,
-    byrow = TRUE
+
+  expect_equal(delay_pmf, complete_pmf, tolerance = 0.001)
+})
+
+test_that(
+  "get_delay_estimate works with every other day reporting of daily data",
+  {
+    sim_delay_pmf <- c(0.1, 0.2, 0.3, 0.1, 0.1, 0.1, 0.1)
+
+    counts <- c(30, 40, 50, 60, 70)
+
+    complete_triangle <- lapply(counts, function(x) x * sim_delay_pmf)
+    complete_triangle <- do.call(rbind, complete_triangle)
+
+    reporting_triangle <- replace_lower_right_with_NA(
+      complete_triangle,
+      structure = 2
+    )
+
+    # Get delay estimate
+    delay_pmf <- get_delay_estimate(
+      reporting_triangle = reporting_triangle
+    )
+    # Test that the function returns the expected PMF
+    expect_identical(delay_pmf, sim_delay_pmf)
+  }
+)
+
+test_that("get_delay_estimate handles complete reporting triangles", {
```

Comment:
By complete do you mean symmetric?

Reply:
I mean complete in the sense you have used in the docs I am not sure what you mean by symmetric?

Reply:
But then you would want? Because the test should make sure that we can get a delay_pmf from a complete reporting matrix. I think I was suggesting "symmetric" meaning 1:1 reference:report because I thought thats what you meant to be testing here. 
```{r}
  delay_pmf <- get_delay_estimate(
    reporting_triangle = partial_complete,
    max_delay = 3,
    n = 4
  )
  ```

Reply:
I see what you mean. The complete matrix step is being tested above. I have clarified that this is a diagonal reporting triangle. Happy to have an issue to discuss nomenclature further perhaps as part of the other suggested renaming.